src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -74,17 +80,17 @@ class ApplyException(orchestrator.OrchestratorError):
     """
 
 
-def threaded(f):
-    def wrapper(*args, **kwargs):
+def threaded(f: FuncT) -> FuncT:
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
```

Comment:
the `wrapper()` returns a `threading.Thread`, i think.

Reply:
that would work with something like 

```diff
diff --git a/src/pybind/mgr/rook/rook_cluster.py b/src/pybind/mgr/rook/rook_cluster.py
index f1f17afd50..ead55f60cc 100644
--- a/src/pybind/mgr/rook/rook_cluster.py
+++ b/src/pybind/mgr/rook/rook_cluster.py
@@ -80,13 +80,13 @@ class ApplyException(orchestrator.OrchestratorError):
     """
 
 
-def threaded(f: FuncT) -> FuncT:
-    def wrapper(*args: Any, **kwargs: Any) -> Any:
+def threaded(f: Callable[..., None]) -> Callable[..., threading.Thread]:
+    def wrapper(*args: Any, **kwargs: Any) -> threading.Thread:
         t = threading.Thread(target=f, args=args, kwargs=kwargs)
         t.start()
         return t
 
-    return cast(FuncT, wrapper)
+    return cast(Callable[..., threading.Thread], wrapper)
 
 
 class KubernetesResource(Generic[T]):
```

but this looses the arguments: See https://stackoverflow.com/questions/65936408/typing-function-when-decorator-change-generic-return-type

Reply:
can we have a yet-another-wrapper around `threaded()` to ensure the return type is correct typed, and keep the hack in the wrapped wrapper.

Reply:
After seeing the answer form @freundTech I went with the `Callable[...` solution.